combat-trainer.lic

```diff
@@ -596,15 +596,23 @@ class LootProcess
     end
     return unless tried_loot
     pause 1
-    if left_hand != pair.first && !@equipment_manager.is_listed_item?(left_hand)
-      echo("out of room, failed to store #{left_hand}")
-      game_state.unlootable(GameObj.left_hand.noun.downcase)
-      dispose_trash(left_hand)
-    end
-    if right_hand != pair.last && !@equipment_manager.is_listed_item?(right_hand)
-      echo("out of room, failed to store #{right_hand}")
-      game_state.unlootable(GameObj.right_hand.noun.downcase)
-      dispose_trash(right_hand)
+    if DRC.left_hand != pair.first || DRC.right_hand != pair.last
+      # Mitigating a race condition when wand-watcher or another script pauses
+      # combat-trainer while it's looting, then resumes in the middle of this function.
+      # Occassionally, the thread hasn't updated the left_hand/right_hand variables
+      # to know what's currently in your hand and will trash what's in your hand!
+      # Do a glance to force the variables to be refreshed before we take action.
+      DRC.bput('glance', 'You glance .*')
+    end
+    if DRC.left_hand != pair.first && !@equipment_manager.is_listed_item?(DRC.left_hand)
+      DRC.message("Out of room, failed to store: #{DRC.left_hand}")
+      game_state.unlootable(DRC.left_hand_noun.downcase)
+      DRCI.dispose_trash(DRC.left_hand)
+    end
+    if DRC.right_hand != pair.last && !@equipment_manager.is_listed_item?(DRC.right_hand)
+      DRC.message("Out of room, failed to store: #{DRC.right_hand}")
+      game_state.unlootable(DRC.right_hand_noun.downcase)
+      DRCI.dispose_trash(DRC.right_hand)
```

Comment:
* Added explicit use of `DRC` and `DRCI` prefixes
* Switch use of `GameObj` with `DRC` to maintain the abstraction from core Lich
* Use `DRC.message` instead of `echo` to display the error